### PR TITLE
Clang-tidy modernize-loop-convert Qt

### LINKED ISSUE #22183: clang-tidy: modernize-loop-convert
Issue to use clang tidy to replace imperative `for` loops with range based for loops where appropriate

### PR DESCRIPTION
**Pull request by TTitcombe**
**Description of work.**
This PR runs clang-tidy on files in the `qt` directory to use range-based for loops where possible.

**To test:**
Code review

Fixes #22183 

*This does not require release notes* because **internal code change*

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/plugins/algorithm_dialogs/src/PeriodicTableWidget.cpp [hunk 1]
```diff
@@ -45,97 +45,95 @@ void PeriodicTableWidget::ColourElements() {
...
+    ColourButton(metalloid, buttonColourStr);
     update();
   }
 }
 void PeriodicTableWidget::ColourNobleGases(
     const QVector<QPushButton *> &nobleGases) {
   QString buttonColourStr = "background-color: rgb(255, 170, 0, 255)";
-  for (auto i = nobleGases.begin(); i != nobleGases.end(); i++) {
-    ColourButton(*i, buttonColourStr);
+  for (auto nobleGase : nobleGases) {
```
Comment (martyngigg) [inline qt/widgets/plugins/algorithm_dialogs/src/PeriodicTableWidget.cpp]: `nobleGas`?

File: qt/widgets/instrumentview/src/BankRenderingHelpers.cpp [hunk 2]
```diff
@@ -494,16 +494,15 @@ void renderStructuredBank(const Mantid::Geometry::ComponentInfo &compInfo,
   for (size_t x = 0; x < colWidth; x += 3) {
     auto index = x / 3;
     const auto &column = compInfo.children(columns[index]);
-    for (size_t y = 0; y < column.size(); ++y) {
-      extractHexahedron(compInfo.shape(column[y]), hex);
-      auto rot = compInfo.rotation(column[y]);
+    for (unsigned long long y : column) {
```
Comment (martyngigg) [inline qt/widgets/instrumentview/src/BankRenderingHelpers.cpp]: Maybe `auto` here?

**Review by martyngigg — CHANGES_REQUESTED** [review]
A vast improvement on the old style loops. Just a few cases where I think we can improve over what `clang-tidy` has done.

File: qt/scientific_interfaces/EnggDiffraction/EnggDiffractionPresenter.cpp [hunk 3]
```diff
@@ -1462,8 +1462,8 @@ std::vector<std::string> EnggDiffractionPresenter::outputFocusTextureFilenames(
   std::vector<std::string> res;
   res.reserve(bankIDs.size());
   std::string prefix = instStr + "_" + runNo + "_focused_texture_bank_";
-  for (size_t b = 0; b < bankIDs.size(); b++) {
-    res.emplace_back(prefix + boost::lexical_cast<std::string>(bankIDs[b]) +
+  for (unsigned long long bankID : bankIDs) {
```
[the remark this answers is not in the document]
Comment (TTitcombe) [reply qt/scientific_interfaces/EnggDiffraction/EnggDiffractionPresenter.cpp]: yep good idea

File: qt/widgets/plugins/algorithm_dialogs/src/PeriodicTableWidget.cpp (hunk 1 above)
Comment (TTitcombe) [reply qt/widgets/plugins/algorithm_dialogs/src/PeriodicTableWidget.cpp]: Ah missed this one- clang-tidy created a variable name by removing the "s" at the end if there is one. Sometimes you get non-words like gase. I'll fix this

Comment (martyngigg) [reply qt/widgets/plugins/algorithm_dialogs/src/PeriodicTableWidget.cpp]: Thanks. I'm generally very impressed at `clang-tidy`s naming choices like deciphering `index` from indices`.

**Comment by martyngigg** [comment]
@TTitcombe Sorry for the delay in getting back to this. Unfortunately there are some merge conflicts now.

**Comment by TTitcombe** [comment]
Thanks, that's been merged now

**Review by martyngigg — APPROVED** [review]

**Pull request merged**